ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/workspace/model/RuntimeImpl.java

```diff
@@ -91,6 +98,14 @@ public String getMachineToken() {
     return warnings;
   }
 
+  @Override
+  public List<CommandImpl> getCommands() {
+    if (commands == null) {
+      commands = new ArrayList<>();
```

Comment:
Is the the commands list supposed to be mutable? `Collectors.toList()` used in the ctor doesn't guarantee that.